Editor/Auditors/BuildAuditor.cs

```diff
@@ -0,0 +1,139 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Unity.ProjectAuditor.Editor.Utils;
+using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+
+namespace Unity.ProjectAuditor.Editor.Auditors
+{
+    public class BuildAuditor : IAuditor, IPostprocessBuildWithReport
+    {
+        static readonly ProblemDescriptor k_Descriptor = new ProblemDescriptor
+            (
+            600000,
+            "Build file",
+            Area.BuildSize
+            );
+
+        static readonly IssueLayout k_IssueLayout = new IssueLayout
+        {
+            category = IssueCategory.BuildFiles,
+            properties = new[]
+            {
+                new PropertyDefinition { type = PropertyType.Description, name = "Source Asset"},
+                new PropertyDefinition { type = PropertyType.FileType, name = "Type"},
+                new PropertyDefinition { type = PropertyType.Custom, format = PropertyFormat.Integer, name = "Size (bytes)", longName = "Size (bytes) in the Build"},
+                new PropertyDefinition { type = PropertyType.Path, name = "Path"},
+                new PropertyDefinition { type = PropertyType.Custom + 1, format = PropertyFormat.String, name = "Build File"}
+            }
+        };
+
+        static BuildReport s_BuildReport;
+
+        const string s_BuildReportDir = "Assets/BuildReports";
+        const string s_LastBuildReportPath = "Library/LastBuild.buildreport";
+
+        public IEnumerable<ProblemDescriptor> GetDescriptors()
+        {
+            yield return k_Descriptor;
+        }
+
+        public IEnumerable<IssueLayout> GetLayouts()
+        {
+            yield return k_IssueLayout;
+        }
+
+        public void Initialize(ProjectAuditorConfig config)
+        {
+        }
+
+        public void RegisterDescriptor(ProblemDescriptor descriptor)
+        {
+        }
+
+        public void Audit(Action<ProjectIssue> onIssueFound, Action onComplete, IProgressBar progressBar = null)
+        {
+#if UNITY_2019_4_OR_NEWER
+            var buildReport = GetBuildReport();
+            if (buildReport != null)
+            {
+                foreach (var packedAsset in buildReport.packedAssets)
+                {
+                    var dict = new Dictionary<GUID, List<PackedAssetInfo>>();
```

Comment:
Could you add a comment to explain why there can be more than one content generated per source asset GUID and why we want to aggregate them together? It's not obvious to me with my limited knowledge of the build process.

Reply:
done: https://github.com/Unity-Technologies/ProjectAuditor/pull/24/commits/390f54b9841757a05cda38551db5948bdff1fcd5

Reply:
Thanks!